rero_ils/modules/stats/api.py

```diff
@@ -687,6 +694,450 @@ def loans_of_transaction_library_by_item_location(self,
+        """
+        trigger = None
+
+        org_pid = extracted_data_from_ref(data['organisation'])
+        org = Organisation.get_record_by_pid(org_pid)
+        library_pids = list(org.get_libraries_pids())
+
+        indicator = data['category'].get('indicator')
+        distributions = indicator.get('distributions')
+
+        if distributions:
+            if len(distributions) == 1:
+                dist1 = distributions[0]
+                dist2 = None
+            elif len(distributions) == 2:
+                if not self._validate_distributions(distributions):
+                    return
+                dist1 = distributions[0]
+                dist2 = distributions[1]
+            else:
+                return
+        else:
+            dist1 = None
+            dist2 = None
+
+        self.config = {'indicator': indicator.get('type'),
+                       'dist1': dist1,
+                       'dist2': dist2,
+                       'filter': indicator.get('filter'),
+                       'period': indicator.get('period'),
+                       'frequency': data['frequency'],
+                       'library_pids': library_pids,
+                       'org_pid': org_pid}
+
+        indicator = self.config["indicator"]
+
+        trigger = self.trigger_mapping.get(indicator)
+
+        if indicator in list(self.trigger_mapping):
+            query = RecordsSearch(index=LoanOperationLog.index_name)[0:0]\
+                    .filter('term', record__type='loan')\
+                    .filter('term', loan__trigger=trigger)
+            return self.number_of(query, trigger)
+        elif indicator == 'number_of_documents':
+            query = DocumentsSearch()[0:0]
+            return self.number_of(query)
+        elif indicator == 'number_of_items':
+            query = ItemsSearch()[0:0]
+            return self.number_of(query)
+        elif indicator == 'number_of_serial_holdings':
+            query = HoldingsSearch()[0:0]\
+                .filter('term', holdings_type='serial')
+            return self.number_of(query)
+        elif indicator == 'number_of_ill_requests':
+            query = ILLRequestsSearch()[0:0]
+            return self.number_of(query)
+        elif indicator == 'number_of_deleted_items':
+            query = RecordsSearch(index=LoanOperationLog.index_name)[0:0]\
+                    .filter('term', record__type='item')\
+                    .filter('term', operation='delete')
+            return self.number_of(query)
+        elif indicator == 'number_of_patrons':
+            query = PatronsSearch()[0:0]
+            return self.number_of(query)
+        # elif indicator == 'number_of_active_patrons':
+        #     pass
+
+        return []
+
+    def create(self, pid, dbcommit=False, reindex=False):
+        """Create report.
+
+        :param pid: report configuration pid
+        """
+        cfg = StatConfiguration.get_record_by_pid(pid)
+        if cfg.get('is_active'):
```

Comment:
In my opinion : If the report isn't active, you should raise an `CustomException`. Inverting the if condition you could save a lot of spaces characters 
```python
if not cfg.get('is_active'):
   raise NotActiveStatConfigException(cfg)
results = self._report_data(cfg)
...
```